resharper/resharper-unity/src/Unity.Shaders/HlslSupport/Integration/Cpp/UnityShaderModuleProvider.cs

```diff
@@ -118,17 +118,10 @@ public string GetPersistentID()
     [SolutionComponent]
     public class UnityShaderPsiModuleProviderFilter : IProjectPsiModuleProviderFilter
     {
-        private readonly UnitySolutionTracker mySolutionTracker;
-
-        public UnityShaderPsiModuleProviderFilter(UnitySolutionTracker solutionTracker)
-        {
-            mySolutionTracker = solutionTracker;
-        }
-
         public Tuple<IProjectPsiModuleHandler, IPsiModuleDecorator> OverrideHandler(Lifetime lifetime, IProject project,
             IProjectPsiModuleHandler handler)
         {
-            if ( handler.PrimaryModule != null && UnityReferencesTracker.ReferencesUnity(project))
+            if ( handler.PrimaryModule != null && project.GetComponent<UnityReferencesTracker>().IsUnityProject(project))
```

Comment:
Can we inject this instead of calling `GetComponent`?

Reply:
Tried that - caused cyclic dependencies, if I remember correct.